src/device.cc

```diff
@@ -81,14 +81,29 @@ static NAN_METHOD(deviceConstructor) {
 	STRUCT_TO_V8(v8dd, dd, bNumConfigurations)
 
 	uint8_t port_numbers[MAX_PORTS];
+	// FreeBSD workarounds based on https://sourceforge.net/p/sigrok/mailman/attachment/54C11DD2.10002%40uffe.org/1/
+#ifdef __FreeBSD__
+	struct libusb_device_handle* dh;
+	CHECK_USB(libusb_open(self->device, &dh));
+#endif
 	int ret = libusb_get_port_numbers(self->device, &port_numbers[0], MAX_PORTS);
+#ifdef __FreeBSD__
+	if (ret <= 0) {
+		Local<Array> array = Nan::New<Array>(1);
+		array->Set(0, Nan::New(libusb_get_device_address(self->device)));
```

Comment:
That's what was done [in the linked sigrok patch](https://sourceforge.net/p/sigrok/mailman/attachment/54C11DD2.10002%40uffe.org/1/).